framework/src/media/HttpInputDataSource.cpp

```diff
@@ -257,8 +257,11 @@ size_t HttpInputDataSource::HeaderCallback(char *data, size_t size, size_t nmemb
 	size_t totalsize = size * nmemb;
 	std::string header(data, totalsize);
 	medvdbg("%s\n", header.c_str());
-	if (header.find(TAG_CONTENT_TYPE) != std::string::npos) {
-		source->mContentType = header.substr(TAG_CONTENT_TYPE.length());
+	auto pos = header.find(TAG_CONTENT_TYPE);
+	if (pos != std::string::npos) {
+		pos = header.find_first_not_of(' ', pos + TAG_CONTENT_TYPE.length());
+		auto end = header.find((char)0x0d, pos); // CR: 0x0d
```

Comment:
Though '\r\n' shall always exist, but do we need to check 'end' for safe?

Reply:
CURL already check the '\r\n'.If not, callback is not called

Reply:
libcurl uses '\n' as delimiter while parsing HTTP header in Curl_http_readwrite_headers.
‘\r’ is not guaranteed here.

Anyway,  we can skip safety check for such really rare malicious HTTP server behavior.